pkg/operator/controller/status/controller.go

```diff
@@ -685,3 +729,45 @@ func operatorStatusesEqual(a, b configv1.ClusterOperatorStatus) bool {
 
 	return true
 }
+
+// joinConditions merges two cluster operator conditions into one.
+// If both conditions have the same status:
+// - Reason becomes: "MultipleComponents[Not]" + condType.
+// - Messages are concatenated.
+// If the new condition has a higher-priority status, it overrides the current one.
+// Status priority (highest to lowest): True, Unknown, False, empty.
+func joinConditions(currCond, newCond configv1.ClusterOperatorStatusCondition, condType string) configv1.ClusterOperatorStatusCondition {
+	switch newCond.Status {
+	case configv1.ConditionTrue:
+		if currCond.Status == configv1.ConditionTrue {
+			currCond.Reason += "And" + newCond.Reason
```

Comment:
If this continues to be Degraded=True for Gateway API CRDs, this doesn't prevent a case of Reason="GatewayAPICRDsDegraded And GatewayAPICRDsDegraded And GatewayAPICRDsDegraded" does it?

Reply:
`computeOperatorDegradedCondition` computes the status from scratch, the current condition from the cluster is not appended with new reasons/messages.

Reply:
Can you make a small unit test for this function that demonstrates that it wouldn't keep appending Reasons everytime a reconcile is made and the Degraded=true condition continues?

Reply:
Well, a unit test for `Reconcile` won't be small, I would need to mock the client and some other things probably.

Can you please explain to me how exactly the Degraded condition can be appended to the existing one? Knowing that `computeOperatorDegradedCondition` does not take [`co.Status.Conditions` ](https://github.com/openshift/cluster-ingress-operator/blob/4eb3a46bb788a9066a647ddfc25792b99fd39f5a/pkg/operator/controller/status/controller.go#L243)as a parameter and that [`mergeConditions` override the condition with the new value](https://github.com/openshift/cluster-ingress-operator/blob/4eb3a46bb788a9066a647ddfc25792b99fd39f5a/pkg/operator/controller/status/controller.go#L682-L692).

Reply:
It's not that the conditions are being appended, it's that the reasons and messages are being appended. Reviewing this again, I see that the condition is blank before the two conditions are joined and it seems this couldn't happen after all.  

I wouldn't want to see more and more degraded reasons and messages being concatenated in the future, two is bad enough in my opinion.  There is no way to enforce this though.  I'll just go on record as not being very imressed with this framework, but I can't override @Miciah's decision so I'll let it go.